src/modules/Dashboard/features/RespondentData/RespondentDataReview/RespondentDataReview.test.tsx

```diff
@@ -0,0 +1,386 @@
+  applet: {
+    applet: {
+      ...initialStateData,
+      data: { result: mockedApplet },
+    },
+  },
+  users: {
+    allRespondents: {
+      ...initialStateData,
+      data: {
+        result: [mockedRespondent, mockedRespondent2],
+        count: 2,
+      },
+    },
+    respondentDetails: {
+      ...initialStateData,
+      data: {
+        result: {
+          nickname: 'Mocked Respondent',
+          secretUserId: mockedRespondentId,
+        },
+      },
+    },
+  },
+};
+
+jest.mock('modules/Dashboard/hooks', () => ({
+  ...jest.requireActual('modules/Dashboard/hooks'),
+  useDecryptedActivityData: jest.fn(),
+}));
+
+jest.mock('modules/Dashboard/features/RespondentData/CollapsedMdText', () => ({
+  __esModule: true,
+  ...jest.requireActual('modules/Dashboard/features/RespondentData/CollapsedMdText'),
+  CollapsedMdText: ({ text }) => <div data-testid="mock-collapsed-md-text">{text}</div>,
+}));
+
+const items = [
+  {
+    question: {
+      en: 'Single Selected - Mocked Item',
+    },
+    responseType: 'singleSelect',
+    responseValues: {
+      paletteName: null,
+      options: [
+        {
+          id: '484596cc-0b4e-42a9-ab9d-20d4dae97d58',
+          text: '1',
+          isHidden: false,
+          value: 0,
+        },
+        {
+          id: 'a6ee9b74-e1d3-47b2-8c7f-fa9a22313b19',
+          text: '2',
+          isHidden: false,
+          value: 1,
+        },
+      ],
+    },
+    config: {
+      removeBackButton: false,
+      skippableItem: true,
+      randomizeOptions: false,
+      timer: 0,
+      addScores: false,
+      setAlerts: false,
+      addTooltip: false,
+      setPalette: false,
+      additionalResponseOption: {
+        textInputOption: false,
+        textInputRequired: false,
+      },
+    },
+    name: 'ss-1',
+    isHidden: false,
+    allowEdit: true,
+    id: 'ab383cc6-834b-45da-a0e1-fc21ca74b316',
+    order: 1,
+  },
+];
+
+const mockedGetWithActivities1 = {
+  data: {
+    result: [
+      {
+        id: '951145fa-3053-4428-a970-70531e383d89',
+        name: 'Activity 1',
+        answerDates: [],
+      },
+    ],
+  },
+};
+
+const mockedGetWithActivities2 = {
+  data: {
+    result: [
+      {
+        id: '951145fa-3053-4428-a970-70531e383d89',
+        name: 'Activity 1',
+        answerDates: [
+          {
+            createdAt: '2023-12-15T11:21:40.509095',
+            answerId: 'ff9e1f86-3fa2-4edd-908c-832810555633',
+          },
+          {
+            createdAt: '2023-12-15T11:22:34.150182',
+            answerId: 'd4147952-73e2-4693-b968-3ecf2468187d',
+          },
+        ],
+      },
+    ],
+  },
+};
+
+const mockedGetWithDates = {
+  data: {
+    result: {
+      dates: ['2023-12-11', '2023-12-15'],
+    },
+  },
+};
+
+const mockedGetWithResponses = {
+  data: {
+    result: {
+      mockedResult: 'mockedResult',
+    },
+  },
+};
+
+describe('RespondentDataReview', () => {
+  test('renders component correctly with all child components', async () => {
```

Comment:
could you split one big test into small separate unit tests, if it applicable?

Reply:
then most of it will be duplicated in each test